### PR TITLE
Added allowance for argument zero_terms_query to be any case. Allowed…

### PR DESCRIPTION
**Pull request by MitchellGale-BitQuill**
Signed-off-by: Mitchell Gale <mitchellg@bitquilltech.com>

### Description
Allows for any case on argument name for relevancy based function (Match Phrase Prefix, Simple Query String,  Match Bool Prefix, Match, Multi-Match, Query String, Relevance, Simple Query String). 
e.g. 
`AnalyZer` can work instead of just `analyzer`
`Zero_Terms_Query` can work instead of just `zero_terms_query`
`BOOst` can work instead of just `boost`

Allows for `zero_term_query` arguments to be of any case.
e.g.
`All` can work instead of `all`
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/RelevanceQuery.java [hunk 1]
```diff
@@ -46,14 +46,15 @@ public QueryBuilder build(FunctionExpression func) {
     Iterator<Expression> iterator = arguments.listIterator(2);
     while (iterator.hasNext()) {
       NamedArgumentExpression arg = (NamedArgumentExpression) iterator.next();
-      if (!queryBuildActions.containsKey(arg.getArgName())) {
+      String argNormalized = arg.getArgName().toLowerCase();
```
Comment (acarbonetto) [inline opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/relevance/RelevanceQuery.java:49]: The isLowerCase may fail if arg.getArgName() doesn't exist.

File: opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/MatchPhraseQueryTest.java [hunk 2]
```diff
@@ -71,6 +71,16 @@ public void test_analyzer_parameter() {
     Assertions.assertNotNull(matchPhraseQuery.build(new MatchPhraseExpression(arguments)));
   }
 
+  @Test
+  public void test_analyzer_parameter_capitalized() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "t1"),
+        namedArgument("query", "t2"),
+        namedArgument("AnalyZER", "standard")
```
Comment (acarbonetto) [inline opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/MatchPhraseQueryTest.java:79]: what about changing the case of `standard`?

File: opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/MatchPhraseQueryTest.java [hunk 3]
```diff
@@ -99,6 +109,26 @@ public void test_zero_terms_query_parameter() {
...
+    );
+    Assertions.assertNotNull(matchPhraseQuery.build(new MatchPhraseExpression(arguments)));
+  }
+
+  @Test
+  public void test_zero_terms_query_parameter_enum_random_case() {
+    List<Expression> arguments = List.of(
+        namedArgument("field", "t1"),
+        namedArgument("query", "t2"),
+        namedArgument("zEro_terMs_QUery", "ALL")
```
Comment (acarbonetto) [inline opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/MatchPhraseQueryTest.java:127]: what about changing case of `ALL`?

**Review by acarbonetto — APPROVED** [review]

File: opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/MatchPhraseQueryTest.java (hunk 3 above)
Comment (MitchellGale-BitQuill) [reply opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/MatchPhraseQueryTest.java:127]: Done in previous test case. Or do you mean doing both at the same time? There is no overlap of the code changes, so I didn't think it was needed to do both changes at the same time.

**Review by Yury-Fridlyand — CHANGES_REQUESTED** [review]
SQL/PPL parser is already case-insensitive. The goal is to change case for some parameter values.
See example from `main`:
```
opensearchsql> select * from logs where simple_query_string(['*'], 'CLR', AnAlYZer = 'standard');
Output longer than terminal width
Do you want to display data vertically for better visual effect? [y/N]: y
opensearchsql> select * from logs where simple_query_string(['*'], 'CLR', analyzer = 'stAndaRd');
TransportError(500, 'SearchPhaseExecutionException', {'error': {'type': 'SearchPhaseExecutionException', 'reason': 'Error occurred in OpenSearch engine: all shards failed', 'details': 'Shard[0]: [logs/uHvAufvyREut9LnosEX2yg] QueryShardException[[simple_query_string] analyzer [stAndaRd] not found]\n\nFor more details, please send request for Json format to see the raw response from OpenSearch engine.'}, 'status': 503})
```
So all calls for `analyzer`'s value should be wrapped by `.toLower()`:
```java
.put("analyzer", (b, v) -> b.analyzer(v.stringValue()))
```
like you did for `ZeroTermsQuery` and `Flags`. Boost has double type, nothing to convert there, but what about `Operator`, `DefaultOperator` and so on?

File: opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/MatchPhraseQueryTest.java (hunk 3 above)
Comment (acarbonetto) [reply opensearch/src/test/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/MatchPhraseQueryTest.java:127]: That's fine.  We can skip.

**Review by MaxKsyunz — CHANGES_REQUESTED** [review]

**Comment by MaxKsyunz** [comment]
@MitchellGale-BitQuill the conflicts need to be resolved and SQL Java CI should pass as well.

**Comment by MitchellGale-BitQuill** [comment]
https://github.com/Bit-Quill/opensearch-project-sql/pull/103 PR remade in 103

**Pull request closed without merging**